### PR TITLE
Support webpack's library stack traces

### LINKED ISSUE #20745: Incorrect "in_app" processing of sourcemaps that have a webpack "namepsace"
## Description

My app's stack frames are marked as non-`in_app` due to the webpack-specific sourcemap processing at https://github.com/getsentry/sentry/blob/d54f3cac09cf6925204a52d43b7ddec510ec3a6b/src/sentry/lang/javascript/processor.py#L695

This code assumes a `webpack:///` prefix, while code that is packaged as a library (or otherwise sets a "namespace") will have a `webpack://[namespace]/` prefix.

See https://webpack.js.org/configuration/output/#outputdevtoolmodulefilenametemplate

### What you expected to happen

Frames should remain marked `in_app`: true

### Possible Solution

Take into account the general case format: `webpack://[optional namespace]/`

┆Issue is synchronized with this [Jira Improvement](https://getsentry.atlassian.net/browse/FEEDBACK-1817) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by ranyitz**
This PR fixes https://github.com/getsentry/sentry/issues/20745 by adding support for a case of webpack library.

### CODE REVIEW

**Comment by ranyitz** [comment]
I'll appreciate your thought on this

File: src/sentry/lang/javascript/processor.py [hunk 1]
```diff
@@ -686,6 +686,7 @@ def process_frame(self, processable_frame, processing_task):
                 if original_function_name is not None:
                     new_frame["function"] = original_function_name
 
+                WEBPACK_WITH_LIBRARY_PREFIX_RE = re.compile(r"webpack:\/\/[\w\-_.]+\/([\w\-_.\/]+)")
```
Comment (kamilogorek) [inline src/sentry/lang/javascript/processor.py:689]: Can we move this regexp compilation webpack branch?

File: src/sentry/lang/javascript/processor.py [hunk 2]
```diff
@@ -697,6 +698,9 @@ def process_frame(self, processable_frame, processing_task):
                     # (i.e. node_modules)
                     if "/~/" in filename:
                         filename = "~/" + abs_path.split("/~/", 1)[-1]
+                    elif WEBPACK_WITH_LIBRARY_PREFIX_RE.search(filename):
+                        # https://github.com/getsentry/sentry/issues/20745
+                        filename = WEBPACK_WITH_LIBRARY_PREFIX_RE.search(filename).group(1)
```
Comment (kamilogorek) [inline src/sentry/lang/javascript/processor.py:703]: We can skip running regexp twice with:
```js
webpack_prefixed = WEBPACK_WITH_LIBRARY_PREFIX_RE.search(filename)
if webpack_prefixed:
    filename = webpack_prefixed.group(1)
```

**Review by kamilogorek — CHANGES_REQUESTED** [review]
Looks good to me, although it'd be great to get a second pair of eyes on this one.

File: src/sentry/lang/javascript/processor.py (hunk 1 above)
Comment (ranyitz) [reply src/sentry/lang/javascript/processor.py]: Moved the regex to the top, let me know if you meant for anything else

File: src/sentry/lang/javascript/processor.py (hunk 2 above)
Comment (ranyitz) [reply src/sentry/lang/javascript/processor.py]: Adjusted to code so it'll run the regex once, thanks

**Comment by ranyitz** [comment]
@kamilogorek Thanks for the review, I made the changes as requested

**Review by kamilogorek — CHANGES_REQUESTED** [review]

**Comment by kamilogorek** [comment]
@ranyitz this was what I meant (so that regexps is not compiled if are not inside `webpack:///` path branch :)
Also changed name to use `namespace` which would be more in line with webpack documentation.

**Comment by ranyitz** [comment]
I see, thanks @kamilogorek!

**Comment by kamilogorek** [comment]
cc @mitsuhiko and maybe @EvanPurkhiser as you're webpack wizard

**Comment by EvanPurkhiser** [comment]
Are there any associated tests we can add for this?

**Comment by ranyitz** [comment]
@EvanPurkhiser I don't mind adding tests, but i'm new to this repository, can you point me to an example test, similar to the one you expect to be written here?

**Pull request closed without merging**